src/core/mac/mac.cpp

```diff
@@ -1037,6 +1049,7 @@ void Mac::TransmitDoneTask(otRadioFrame *aFrame, otRadioFrame *aAckFrame, otErro
     case kStateTransmitData:
         if (framePending)
         {
+            mWaitingForData = true;
```

Comment:
We should only execute this block this if the transmit frame is a IEEE 802.15.4 Data Request command frame.